bcipy/signal/tests/model/test_pca_rda_kde.py

```diff
@@ -195,5 +189,75 @@ def test_rda(self):
         self.assertTrue(np.allclose(z, expected))
 
 
+class TestPcaRdaKdeModelExternals(SharedSetup):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass
+
+    def setUp(self):
+        np.random.seed(0)
+        self.model = PcaRdaKdeModel(k_folds=10)
+        self.model.fit(self.x, self.y)
+        np.random.seed(0)
+
+    @pytest.mark.mpl_image_compare(
+        baseline_dir=expected_output_folder, filename="test_inference.expected.png", remove_text=True
+    )
+    def test_fit_predict(self):
+        """Fit and then predict"""
+        alp = alphabet()
+
+        # Create test items that resemble the fake training data
+        num_x_p = 1
+        num_x_n = 9
+
+        x_test_pos = self.pos_mean + self.pos_std * np.random.randn(self.num_channel, num_x_p, self.dim_x)
+        x_test_neg = self.neg_mean + self.neg_std * np.random.randn(self.num_channel, num_x_n, self.dim_x)
+        x_test = np.concatenate((x_test_pos, x_test_neg), 1)  # Target letter is first
+
+        letters = alp[10 : 10 + num_x_p + num_x_n]  # Target letter is K
+
+        lik_r = self.model.predict(data=x_test, inquiry=letters, symbol_set=alp)
+        fig, ax = plt.subplots()
+        ax.plot(np.arange(len(alp)), lik_r, "ro")
+        ax.set_xticks(np.arange(len(alp)))
+        ax.set_xticklabels(alp)
+        ax.set_yticks(np.arange(0, 101, 10))
+        return fig
+
+    def test_evaluate(self):
+        model_report = self.model.evaluate(self.x, self.y)
+        self.assertEqual(model_report, ModelEvaluationReport(1.0))
+
+    def test_save_load(self):
```

Comment:
Let's separate the load and save tests

Reply:
I wasn't sure how to actually test the saving/loading other than just checking "the model behavior is the same". The contents of the model itself don't seem to have a useful `__eq__`, even if you keep digging deeper until you get to the scipy library functions being used. Would be happy to change this but I think I could use some help on figuring out how to compare the models.

Reply:
I'm gonna leave this for now unless you have ideas on how to test this.

Note that the previous test of loading was not testing whether a classifier is loaded correctly, but rather just whether a `parameters` dict was successfully pickled and unpickled, so the question of "How can we compare two model objects?" has not been previously answered as far as I can see.

I can't think of a simple way to compare these model objects (which are wrappers around relatively complicated Scikit-Learn objects) other than testing their behavior on some random inputs. Basically an equality comparison would have to go through each model component, identify all sklearn components inside, identify the parts of these which are modified during training and how to compare them. 

I think that may be the reason these objects like (https://github.com/scikit-learn/scikit-learn/blob/15a949460/sklearn/decomposition/_pca.py#L114 and https://github.com/scikit-learn/scikit-learn/blob/15a949460/sklearn/neighbors/_kde.py#L27) don't already have `__eq__` in sklearn.

Reply:
I wish we chatted in person but I was thinking to assert save and load behavior separately. You can mock the save / pkl and assert the necessary things were called and/or assert a file as expected was written (not that it loads). In another test you can mock a .pkl file and assert it did what it was supposed to on load. This will catch issues or changes in the individual methods. The test_save_load is a more integrated test of functionality where the comparison as you mention is important. I think your approach was fine. The fewer things you assert (ideally only one per test) and methods of interest you call per unittest the better as a rule of thumb. I won't block for this.